yabt/test_utils.py

```diff
@@ -31,11 +35,244 @@ def generate_random_dag(nodes, min_rank=0, max_rank=10, edge_prob=0.3):
+                    "MinChilds": 0
+                },
+                {
+                    "Name": "PythonTest",
+                    "Childs": ["PythonPackage", "Python"],
+                    "MinChilds": 2
+                }
+            ]
+        },
+        {
+            "Config": {
+                "MinNameLen": 2, "MaxNameLen": 5,
+                "Steps": 100, "MaxStepNodes": 8
+            }
+        }
+    ]
+
+    class CBuildTrgtType:
+        """ Type of project"""
+        def __init__(self, trg):
+            self.name = trg['Name']
+            self.childs = trg['Childs']  # Types of available dependencies
+            self.min_childs = trg['MinChilds']  # Minimum of dependencies
+
+        def is_child(self, trg):
+            return trg.name in self.childs
+
+    def __init__(self):
+        bd_trgs = CBuildTrgtTest.sort_test_cfg[0]['BuildTargets']
+        self.trgs = list()
+        for trg in bd_trgs:
+            self.trgs.append(CBuildTrgtTest.CBuildTrgtType(trg))
+        cfg = CBuildTrgtTest.sort_test_cfg[1]['Config']
+        self.min_name_len = cfg['MinNameLen']
+        self.max_name_len = cfg['MaxNameLen']
+        self.steps = cfg['Steps']
+        self.max_step_nodes = cfg['MaxStepNodes']
+        self.letters = string.ascii_lowercase
+
+    def __new_build_trgt(self):
+        """Generate a random string """
+        str_len = random.randint(self.min_name_len, self.max_name_len)
+        name = ''.join(random.choice(self.letters) for i in range(str_len))
+        i = random.randint(0, len(self.trgs) - 1)
+        return [name, self.trgs[i]]
+
+    def __add_nodes(self, nx_g):
+        """ Added random nodes with its random dependencies"""
+        d_t = dict()
+        i_n = random.randint(1, self.max_step_nodes)
+        for _ in range(i_n):
+            trg = self.__new_build_trgt()
+            d_t.update({trg[0]: trg[1]})
+        l_g = list(nx_g)
+        l_d = nx.get_node_attributes(nx_g, 'trg')
+        l_n = len(l_g)
+        for name in d_t:
+            if name not in nx_g:
+                trg = d_t[name]
+                childs = 0
+                if l_n > 0:
+                    r_n = random.randint(0, l_n - 1)
+                    for i in range(0, r_n):
+                        child_name = l_g[i]
+                        child_trg = l_d[child_name]
+                        if trg.is_child(child_trg):
+                            nx_g.add_edge(name, child_name)
+                            childs = childs + 1
+                if trg.min_childs <= childs:
+                    nx_g.add_node(name, trg=trg)
+                elif childs > 0:
+                    nx_g.remove_node(name)
+
+    def create_rand_graph(self):
+        """ Created string random DAG """
+        nx_g = nx.DiGraph()
+        for _ in range(self.steps):
+            self.__add_nodes(nx_g)
+        return nx_g
+
+
+def calc_node_hash(graph, sort_g_list, node):
```

Comment:
This seems to be related specifically to your test, then it should be in your test file and not here